tekton/mario-bot/mario-image-build-trigger.yaml

```diff
@@ -149,69 +139,126 @@ spec:
-              params:
-              - name: url
-                value: $(tt.params.targetImage)
-          - name: endtrigger
-            resourceRef:
-              name: github-feedback-trigger
+      - name: gitRepository
+        value: $(tt.params.gitRepository)
+      - name: gitRevision
+        value: $(tt.params.gitRevision)
+      - name: targetImage
+        value: $(tt.params.targetImage)
+      - name: contextPath
+        value: $(tt.params.contextPath)
+      - name: pullRequestID
+        value: $(tt.params.pullRequestID)
+      - name: buildUUID
+        value: $(tt.params.buildUUID)
+---
+apiVersion: tekton.dev/v1beta1
+kind: Pipeline
+metadata:
+  name: clone-and-build
+spec:
+  params:
+  - name: gitRepository
+  - name: gitRevision
+  - name: gitCloneDepth
+    default: "10"
+  - name: targetImage
+  - name: contextPath
+  - name: cloudEventSink
+    default: http://el-github-feedback-trigger.mario:8080
+  - name: pullRequestID
+  - name: buildUUID
+  workspaces:
+  - name: source-code
+  - name: gcp-secret
+  tasks:
+  - name: clone
+    taskRef:
+      name: git-clone
+      bundle: gcr.io/tekton-releases/catalog/upstream/git-clone:0.4
+    params:
+    - name: url
+      value: https://$(params.gitRepository)
+    - name: revision
+      value: $(params.gitRevision)
+    - name: depth
+      value: $(params.gitCloneDepth)
+    workspaces:
+    - name: output
+      workspace: source-code
+  - name: build
+    runAfter:
+      - clone
+    params:
+    - name: contextPath
+      value: $(params.contextPath)
+    - name: targetImage
+      value: $(params.targetImage)
+    workspaces:
+    - name: source-code
+      workspace: source-code
+    - name: gcp-secret
+      workspace: gcp-secret
+    taskSpec:
+      params:
+      - name: contextPath
+      - name: targetImage
+      workspaces:
+      - name: source-code
+      - name: gcp-secret
+        mountPath: /secret
+      steps:
+      - name: build-and-push
+        workingdir: $(workspaces.source-code.path)
+        image: gcr.io/kaniko-project/executor:v0.13.0
+        env:
+        - name: GOOGLE_APPLICATION_CREDENTIALS
+          value: /secret/release.json
+        command:
+        - /kaniko/executor
+        - --dockerfile=Dockerfile
+        - --context=$(params.contextPath)
+        - --destination=$(params.targetImage)
+  finally:
+  - name: publish-event
+    params:
+    - name: sink
+      value: $(params.cloudEventSink)
+    - name: eventID
+      value: $(context.taskRun.name)
+    - name: eventType
+      value: "dev.tekton.event.task.$(tasks.build.status)"
+    - name: data
+      value: '{"pull-request-id": "$(params.pullRequestID)", "build-id": "$(params.buildUUID)", "git-url": "https://$(params.gitRepository)", "git-revision": "$(params.gitRevision)", "target-image": "$(params.targetImage)", "status": "$(tasks.build.status)"}'
+    taskRef:
+      name: cloudevent
+      bundle: gcr.io/tekton-releases/catalog/upstream/cloudevent:0.1
```

Comment:
This could also be re-written using native cloud-events. 
Those are all sent to a central event listener, so the trigger would have to be added there.
It's nice though to use `finally` in a dogfooding pipeline :)

Reply:
I think I'll leave as is for now both because there's a lot changing in this PR and to use `finally`, but this is a good idea for the future!